stl/inc/ranges

```diff
@@ -1641,7 +1643,16 @@ namespace ranges {
                 return _Last;
             }
 
-            _NODISCARD friend constexpr bool operator==(const _Counted_Iter& _Left, const _Sentinel& _Right) {
+            _NODISCARD friend constexpr bool operator==(const _Counted_Iter<_Const>& _Left, const _Sentinel& _Right) {
+                return _Left.count() == 0 || _Left.base() == _Right._Last;
+            }
+
+            // clang-format off
+            template <bool _OtherConst = !_Const>
+                requires sentinel_for<_Base_sentinel, _Counted_Iter<_OtherConst>>

```

Comment:
The resolution depicts `requires sentinel_for<sentinel_t<Base>, iterator_t<maybe-const<OtherConst, V>>>`.

* I see that `_Base_sentinel` is `_Maybe_wrapped<_Wrapped, sentinel_t<_Base_Ty>>`, so that seems to match `sentinel_t<Base>` with our wrapping logic.
* But `_Counted_Iter<_OtherConst>` is `counted_iterator<_Maybe_wrapped<_Wrapped, iterator_t<_Maybe_const<_OtherConst, _Vw>>>>`, which applies `counted_iterator` on top of the wrapping and `iterator_t<maybe-const<OtherConst, V>>` depicted in the Standardese.

I'm not enough of a ranges expert to determine whether this is a problem, or a convenient shortcut.

Reply:
Yip: this should be `sentinel_for<_Base_sentinel, _Maybe_wrapped<_Wrapped, iterator_t<_Maybe_const<_OtherConst, _Vw>>>>` AFAICS. Fixing.